src/components/atom/Modal/index.tsx

```diff
@@ -1,15 +1,18 @@
 import { FC, ReactNode, Fragment } from "react";
 import { Dialog, Transition } from "@headlessui/react";
 import { ReactComponent as ModalCloseIcon } from "../../../assets/img/modal-close-icon.svg";
+import { ReactComponent as BackArrow } from "../../../assets/img/back-arrow.svg";
 
 interface ModalProps {
   isOpen: boolean;
   title?: string;
-  onClose: () => void;
   children: ReactNode;
+  showBackButton?: boolean;
+  onClose: () => void;
+  onBack?: () => void;
 }
 
-const Modal: FC<ModalProps> = ({ isOpen, onClose, children, title }) => {
+const Modal: FC<ModalProps> = ({ isOpen, children, title, showBackButton = false, onClose, onBack }) => {
```

Comment:
showBackButton is redundant here. Instead of checking showBackButton below u can check whether is onBack called. So if u have onBack that means show the button